solr/contrib/ltr/src/java/org/apache/solr/ltr/LTRScoringQuery.java

```diff
@@ -210,7 +208,7 @@ public ModelWeight createWeight(IndexSearcher searcher, boolean needsScores, flo
     final FeatureWeight[] modelFeaturesWeights = new FeatureWeight[modelFeatSize];
     List<FeatureWeight > featureWeights = new ArrayList<>(features.size());
     
-    if(ltrThreadMgr.getMaxThreads() <= 1 || ltrThreadMgr.getMaxQueryThreads() <= 1){
+    if (ltrThreadMgr == null) {
```

Comment:
We dont want the overhead of parallel weight creation if they only specify 0/1, which is what the checks were for before.  If you want to check null here, push the old checks up into the constructor instead.